File: Backend/UtilityBillingSystem/UtilityBillingSystem/Models/Core/Payment.cs
using System.ComponentModel.DataAnnotations;

namespace UtilityBillingSystem.Models.Core
{
    public class Payment
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        [Required]
        public string BillId { get; set; } = string.Empty;
        public DateTime PaymentDate { get; set; } = DateTime.UtcNow;
        
        [Required]
        [Range(0, double.MaxValue, ErrorMessage = "Amount must be non-negative")]
        public decimal Amount { get; set; }
        
        [Required]
        [MaxLength(20)]
        public string PaymentMethod { get; set; } = string.Empty; // Cash, Online
        public string? ReceiptNumber { get; set; }
        public string? UpiId { get; set; }
        
        [MaxLength(20)]
        public string Status { get; set; } = "Completed"; // For future extension

        // Navigation
        public Bill Bill { get; set; } = null!;
    }
}

File: Frontend/UtilityBillingSystem/src/app/models/payment.ts
export interface PaymentRequestDto {
  paymentMethod: string;
  receiptNumber?: string;
  upiId?: string;
}

export interface PaymentHistoryFiltersDto {
  startDate?: Date | null;
  endDate?: Date | null;
  utilityTypeId?: string | null;
}

export interface PaymentHistoryItem {
  id: string;
  paymentDate: string;
  billId: string;
  billingPeriod: string;
  utilityName: string;
  amount: number;
  paymentMethod: string;
  status: string;
}

